distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Aggregator.enso

```diff
@@ -0,0 +1,149 @@
+from Standard.Base import all
+
+from Standard.Table.Data.Column as Column_Module import Column
+import Standard.Table.Data.Group_By_Key
+from Standard.Table.Data.Aggregate_Column import all
+
+## Type used to compute an aggregate column
+type Aggregate_Column_Aggregator
+    type Aggregate_Column_Aggregator initial accumulator finalizer
+
+## Given a Table and a Column create an aggregator
+new : Table->Aggregate_Column->Aggregate_Column_Aggregator
+new table column =
+    create_closure c function:(Column->Any->Integer->Any) = function (column.resolve_column table c)
+
+    is_empty s = if s.is_nothing then True else case s of
+        Text -> s.is_empty
+        _ -> Error.throw (Invalid_Aggregation_Method this.col "Empty is only valid for Text")
+
+    create_aggregator initial:Any=Nothing (accumulator:(Any->Integer->Any)=c->_->c) (finalizer:(Any->Any)=(v->v)) =
+        Aggregate_Column_Aggregator initial accumulator finalizer
+
+    ## Given a map of values and counts, find the value at a specified percentile
+    percentile p:Decimal value:Map =
+        count = value.fold 0 (+)
+        if count == 0 then Nothing else
+            mid_value = (count - 1)*p + 1
+            if mid_value <= 1 then value.first.first else
+                if mid_value >= count then value.last.first else
+                    mid = mid_value.floor
+                    output = value.fold_with_key [0, Nothing, Nothing] c->k->v->
+                        new_v = c.first + v
+                        new_s = if c.first.up_to new_v . contains (mid-1) then k else c.second
+                        new_t = if c.first.up_to new_v . contains mid then k else (c.at 2)
+                        [new_v, new_s, new_t]
+                    (output.second + (output.at 2 - output.second) * (mid_value - mid))
```

Comment:
Maybe this could be moved outside to be a separate method? Otherwise the body of `new` is quite huge. We should just mark it as `PRIVATE`.